nixos/modules/services/networking/gobgpd.nix

```diff
@@ -0,0 +1,73 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gobgpd;
+  format = pkgs.formats.toml { };
+in {
+  options.services.gobgpd = {
+    enable = mkEnableOption "Whether to enable GoBGP Routing Daemon.";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.gobgpd;
+      defaultText = "pkgs.gobgpd";
+      description = ''
+        The package to use.
+      '';
+    };
+
+    settings = mkOption {
+      type = format.type;
+      default = { };
+      description = ''
+        GoBGP configuration. Refer to
+        <link xlink:href="https://github.com/osrg/gobgp#documentation"/>
+        for details on supported values.
+      '';
+      example = literalExample ''
+        {
+          global = {
+            config = {
+              as = 64512;
+              router-id = "192.168.255.1";
+            };
+          };
+          neighbors = [
+            {
+              config = {
+                neighbor-address = "10.0.255.1";
+                peer-as = 65001;
+              };
+            }
+            {
+              config = {
+                neighbor-address = "10.0.255.2";
+                peer-as = 65002;
+              };
+            }
+          ];
+        }
+      '';
+    };
+  };
+
+  config = mkIf (cfg.enable) {
+    environment.systemPackages = [ cfg.package ];
+    environment.etc."gobgpd.conf".source = format.generate "gobgpd.conf" cfg.settings;
```

Comment:
Some people strongly discourage placing configuration files under `/etc` unless the software absolutely demands it. The alternative being to specify the configuration file location in `ExecStart` instead, of course.

Reply:
Changed to generated configuration file path instead of under `/etc`.